plugins/core/core/src/software/aws/toolkits/core/utils/RemoteResourceResolver.kt

```diff
@@ -84,6 +118,30 @@ class DefaultRemoteResourceResolver(
         return expectedLocation
     }
 
+    private fun updateETags(eTagProvider: ETagProvider, endpoint: String): Boolean {
+        val currentEtag = eTagProvider.etag
+        val remoteEtag = getEndpointETag(endpoint)
+        eTagProvider.etag = remoteEtag
+        return currentEtag != remoteEtag
+    }
+
+    private fun getEndpointETag(endpoint: String): String =
+        try {
+            val url = URI(endpoint).toURL()
+            (url.openConnection() as HttpURLConnection).let { connection ->
```

Comment:
java 11 has a newer way of doing this, but let's actually use the HttpRequests util you were using in a previous PR